modules/standard/Socket.md

```diff
@@ -0,0 +1,136 @@
+## Socket Module API
+
+To those familiar with the Unix socket API, the method names will feel familiar, though their usage will be somewhat simpler than the raw Unix socket API.
+
+## Connecting to remote sockets
+
+`connect` function will create `TCPConn` instance which will take in the following parameters
+
+**Method:**
+```chapel
+connect(addr:IPAddr,port:int) -> TCPConn
+```
+
+**Parameters:**
+
+```
+addr: IPAddr - ipv4/ipv6 address
+port: Integer - port number
+```
+
+A higher level construct for socket will take in addr as `string` which will we resolved and converted to `IPAddr` internally.
+
+**Method:**
+```chapel
+connect(addr:string,port:int) -> TCPConn
+```
+
+**Parameters:**
+```
+addr: String - host name either in dot-dash or domain form
+port: Integer - port number
+```
+
+The `TCPConn` instance will have methods on it to call
+- `read` ,
+- `write` ,
+- `close`,
+- `closeWrite`,
+- `closeRead`,
+- `setKeepAlive`,
+- `addr` ,
+- `setReadBuffer`
+- and `setWriteBuffer`
+
+The additional support with `TCPConn` and `TCPListener` will be to enable or disable two of the prominent optimization algorithm for TCP Protocol that are:
+
+- `nagle`
+- `quickack`
+
+The support will be in form of separate utility function attached with socket module that can set/unset the optimization based on `boolean` argument
+
+## Creating a Listener
+
+`listen` function will create a `TCPListener` Instance which will take in following parameters
+
+**Method:**
+```chapel
+listen(addr:IPAddr,port:int,backlog=5,reuseaddr=true) -> TCPListener
+```
+
+**Parameters**
+```
+addr: IPAddr - ipv4/ipv6 address
+port: Integer - port number
+backlog: Integer - optional parameter, default value = 0
+reuseaddr: Boolean - optional parameter, default value = true
+```
+
+The `TCPListener` instance will have methods on it to call
+- `accept`,
+- `close`,
+- `addr` and
+- `setSocketOpt` to set various socket options.
```

Comment:
Why does TCPConn use setKeepAlive etc but TCPListener uses setSocketOpt? Seems like the approach for setting the socket options should be consistent between the two.

Reply:
`setKeepAlive` is something that we can't associate with a socket in listening mode so it isn't available on `TCPListener`. I wasn't sure what will be the options user might want to set on a `TCPListener` there are several [list](https://www.ibm.com/docs/en/zos/2.3.0?topic=functions-setsockopt-set-options-associated-socket) so I associated the default `setSocketOpt` function with `TCPListener`

Reply:
I have updated it to just `setSocketOpt` because if we aren't using C IO then providing options like `RECVBUF` wont translate to chapel IO